src/common/SegmentedButton.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import { ButtonStyle, getThemeColor, getReverseThemeColor } from "./Button";
+import ActionLink from './ActionLink';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCertificate } from "@fortawesome/free-solid-svg-icons";
+import Color from 'color';
+import classnames from "classnames";
+
+const SegmentedButton: React.FC<SegmentedButtonProps> = (props) => {
+  const [selected, setSelected] = useState(props.selected);
+  const THEME_COLOR = getThemeColor(props.theme);
+  const REVERSE_THEME_COLOR = getReverseThemeColor(props.theme);
+  const transparent = ButtonStyle.TRANSPARENT == props.theme;
+  return (
+    <div className={classnames("segmented-button", {})}>
+      {props.options.map((option) => (
+        <div className={classnames("segmented-button-option", selected == option.id ? "selected" : {})} key={"" + option.id}>
+          {option.updates && (
+            <div className="updates">
+              {option.updates === true ? (
+                <FontAwesomeIcon icon={faCertificate} />
+              ) : option.updates == Infinity ? (
+                "∞"
+              ) : (
+                option.updates
+              )}
+            </div>
+          )}
+          <ActionLink
+            link={{ 
```

Comment:
This is a slightly annoying thing and it doesn't matter that much, but passing an object this way causes `ActionLink` to re-render every single time because the link object is always different. That said, because you're in a loop, you can't simply use `useMemo` directly here to get a consistent object, since it will complain that your number of hooks are different if the number of options ever changes from one render to the other.

There's a couple ways of fixing this: 1) separate the option in its own render class `"SegmentedButtonOption"` that renders a single option, so you can use `useCallback` (this is likely to be easier) and 2) use `useMemo` to get an array of handlers for the various options, and only recalculate that when `props.options` changes (this might or might not cause eslint to yell at you about your re-rendering conditions sucking).

You don't have to fix this, just letting you know it's a thing. I think the consensus is that avoiding re-rendering is not necessary unless those re-renderings are expensive, and I don't think they are here. That said, sometimes I go on "let's remove unnecessary re-renders" fixing sprees because I like when my performance analyses don't show stuff changing unnecessarily.

Reply:
Wow, thanks so much! I have lots to learn about React's render flow. I'll try the first method you mentioned.